cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -745,6 +745,17 @@ void isolated_helper_function(...);
 
 [**Anonymous namespaces should *never* be used in a header file.**](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file)
 
+# Deprecating and Removing Code
+
+If changes to libcudf involve breaking existing APIs, libcudf follows the following policy:
+- Code should be deprecated for one release prior to removal.
+- Deprecations should be indicated using the [deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated) attribute.
+- Deprecated APIs should be documented using Doxygen's [deprecated](https://www.doxygen.nl/manual/commands.html#cmddeprecated) command.
```

Comment:
This technically should go in DOCUMENTATION.md, not here, but I figured it was a single line and would be more discoverable here. I can add something to DOCUMENTATION.md in addition (or instead) and link here if reviewers think there would be a benefit.

Reply:
I think it should go in both. Please also add an example of using the doxygen `deprecated` command to DOCUMENTATION.md

Reply:
Done. I also added it to an API locally and built the docs to verify that it worked as expected. Conveniently, in addition to showing up on the documentation for a particular API it looks like doxygen will generate a page containing all deprecations:

![image](https://user-images.githubusercontent.com/1538165/154392371-cea3166d-8f6d-4314-a425-a4e7116f6362.png)


![image](https://user-images.githubusercontent.com/1538165/154392324-4a201c76-d626-4e5f-8cee-c04ac2ad4cd5.png)